modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/management_bar/components/filters/DynamicFilter.js

```diff
@@ -0,0 +1,281 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayButton from '@clayui/button';
+import ClayDropDown from '@clayui/drop-down';
+import ClayMultiSelect from '@clayui/multi-select';
+import {fetch, openToast} from 'frontend-js-web';
+import PropTypes from 'prop-types';
+import React, {useEffect, useState} from 'react';
+
+const getSelectedItemsLabel = ({selectedData}) => {
+	return selectedData.map((item) => item.label).join(', ');
+};
+
+const getOdataString = ({id, selectedData}) => {
+	return `${id} in (${selectedData
+		.map((item) => `'${item.value}'`)
+		.join(',')})`;
+};
+
+async function getPicklist(erc) {
+	const response = await fetch(
+		`/o/headless-admin-list-type/v1.0/list-type-definitions/by-external-reference-code/${erc}`
+	);
+
+	if (!response.ok) {
+		openToast({
+			message: Liferay.Language.get('your-request-failed-to-complete'),
+			type: 'danger',
+		});
+
+		return [];
+	}
+
+	return await response.json();
+}
+
+const DynamicFilter = ({
+	id,
+	listTypeDefinitionId,
+	multiple,
+	preselectedValues,
+	selectedData,
+	setFilter,
+}) => {
+	const [selectedItems, setSelectedItems] = useState(
+		selectedData ? selectedData : preselectedValues
+	);
+	const [sourceItems, setSourceItems] = useState([]);
+	const [availableItems, setAvailableItems] = useState([]);
+
+	useEffect(() => {
+		getPicklist(listTypeDefinitionId).then((picklist) => {
+			const {listTypeEntries} = picklist;
+
+			const availableItems = listTypeEntries.map((entry) => ({
+				label:
+					entry.name_i18n[Liferay.ThemeDisplay.getLanguageId()] ||
+					entry.name,
+				value: String(entry.key),
+			}));
+
+			setAvailableItems(availableItems);
+		});
+	}, [listTypeDefinitionId]);
+
+	useEffect(() => {
+		if (multiple) {
+			const selectedValues = selectedItems.map((item) => item.value);
+
+			setSourceItems(
+				availableItems.filter(
+					(item) => !selectedValues.includes(item.value)
+				)
+			);
+		}
+		else {
+			setSourceItems(selectedItems.length ? [] : availableItems);
+		}
+	}, [availableItems, multiple, selectedItems]);
+
+	return (
+		<>
+			<ClayDropDown.Caption>
+				<ClayMultiSelect
+					allowsCustomLabel={false}
+					items={selectedItems}
+					loadingState={4}
+					onItemsChange={(items) => {
+						const presentValues = {};
+
+						const selectedItems = items.filter((item) => {
+							if (presentValues[item.value]) {
+								return false;
+							}
+
+							presentValues[item.value] = true;
+
+							return true;
+						});
+
+						setSelectedItems(selectedItems);
+					}}
+					sourceItems={sourceItems}
+					value=""
+				/>
+			</ClayDropDown.Caption>
+			<ClayDropDown.Divider />
+			<ClayDropDown.Caption>
+				<ClayButton
```

Comment:
I think you probably need some aria stuff on this component for accessibility.

Reply:
:thinking: 

Problem 1 is I have no idea about accessibility since I don't usually write UI code :cry: 
Problem 2 is I copied this from another place (`DateRangeFilter.js`) :grimacing: